contrib/pkg/awsprivatelink/endpointvpc/remove.go

```diff
@@ -0,0 +1,353 @@
+				o.endpointVpcClients,
+				aws.String(endpointVPCDefaultSG),
+				aws.String(associatedVpcWorkerSG),
+			); err != nil {
+				// Proceed if ingress not found, fail otherwise
+				switch aerr, ok := err.(awserr.Error); {
+				case ok && aerr.Code() == "InvalidPermission.NotFound":
+					log.Warnf("Access from the associated VPC's worker SG to the endpoint VPC's default SG is not enabled")
+				default:
+					log.WithError(err).Fatal("Failed to revoke access from the associated VPC's worker SG to the endpoint VPC's default SG")
+				}
+			}
+
+		// Associated VPC & endpoint VPC in different regions => revoke ingress from CIDR of the peer
+		default:
+			log.Info("Revoking access from the endpoint VPC's CIDR block to the associated VPC's worker SG")
+			if _, err = awsutils.RevokeAllIngressFromCIDR(
+				associatedVpcClients,
+				aws.String(associatedVpcWorkerSG),
+				aws.String(endpointVpcCIDR),
+			); err != nil {
+				// Proceed if ingress not found, fail otherwise
+				switch aerr, ok := err.(awserr.Error); {
+				case ok && aerr.Code() == "InvalidPermission.NotFound":
+					log.Warnf("Access from the endpoint VPC's CIDR block to the associated VPC's worker SG is not enabled")
+				default:
+					log.WithError(err).Fatal("Failed to revoke access from the endpoint VPC's CIDR block to the associated VPC's worker SG")
+				}
+			}
+
+			log.Info("Revoking access from the associated VPC's CIDR block to the endpoint VPC's default SG")
+			if _, err = awsutils.RevokeAllIngressFromCIDR(
+				o.endpointVpcClients,
+				aws.String(endpointVPCDefaultSG),
+				aws.String(associatedVpcCIDR),
+			); err != nil {
+				// Proceed if ingress not found, fail otherwise
+				switch aerr, ok := err.(awserr.Error); {
+				case ok && aerr.Code() == "InvalidPermission.NotFound":
+					log.Warnf("Access from the associated VPC's CIDR block to the endpoint VPC's default SG is not enabled")
+				default:
+					log.WithError(err).Fatal("Failed to revoke access from the associated VPC's CIDR block to the endpoint VPC's default SG")
+				}
+			}
+		}
+	}
+
+	// Update HiveConfig
+	o.removeEndpointVpcFromHiveConfig()
+
+	return nil
+}
+
+func (o *endpointVPCRemoveOptions) removeEndpointVpcFromHiveConfig() {
+	log.Infof("Removing endpoint VPC %v from HiveConfig", o.endpointVpcId)
+
+	// Remove endpoint VPC from HiveConfig if necessary
+	o.hiveConfig.Spec.AWSPrivateLink.EndpointVPCInventory[o.endpointVpcIdx] =
+		o.hiveConfig.Spec.AWSPrivateLink.EndpointVPCInventory[len(o.hiveConfig.Spec.AWSPrivateLink.EndpointVPCInventory)-1]
+	o.hiveConfig.Spec.AWSPrivateLink.EndpointVPCInventory =
+		o.hiveConfig.Spec.AWSPrivateLink.EndpointVPCInventory[:len(o.hiveConfig.Spec.AWSPrivateLink.EndpointVPCInventory)-1]
+	if err := o.dynamicClient.Update(context.Background(), &o.hiveConfig); err != nil {
+		log.WithError(err).Error("Failed to update HiveConfig")
+	}
+}
+
+func deleteVpcPeeringConnection(awsClients awsclient.Client, VpcId1, VpcId2 *string) (*string, error) {
```

Comment:
I noticed the first return value is not used, might as well get rid of it.

Reply:
Yep, fixed.